WooCommerce/Classes/ViewRelated/Products/Variations/ProductVariationsViewModel.swift

```diff
@@ -39,6 +39,13 @@ final class ProductVariationsViewModel {
             let variationsToGenerate = ProductVariationGenerator.generateVariations(for: product, excluding: existingVariations)
             print("Variations to Generate: \(variationsToGenerate.count)")
 
+            // Guard for 100 variation limit
```

Comment:
Can't we guard for this overflow early? So we won't actually create a lot of objects before failing? We should be able to quickly count possible combinations by multiplying the amount of options for each attribute.

But I'm not sure if we can easily subtract a number of existing variations - there may be duplicated combinations. Can we subtract just total number of existing variations for pre-generation guard (for a worst case)? And if it <100, we can proceed to existing check to make additional verification of combinations.

Reply:
I don't think we will hit a performance problem here right now. The plan is to eventually ask core to generate variations so probably not worth spending extra time on performance unless we notice something significant!